manager/src/main/java/io/gingersnapproject/Caches.java

```diff
@@ -70,6 +70,18 @@ public Uni<String> put(String name, String key, String value) {
       return indexUni;
    }
 
+   public Uni<String> putAll(String name, Map<String, String> values) {
+      Uni<String> bulkIndexingOperation = indexingHandler.putAll(name, values)
+            .memoize().indefinitely();
+
+      for (Map.Entry<String, String> entry : values.entrySet()) {
+         getOrCreateMap(name)
+               .put(entry.getKey(), Uni.createFrom().item(entry.getValue()));
```

Comment:
```suggestion
               .put(entry.getKey(), UniItem.fromItem(entry.getValue()));
```
We have to make sure any completed value is a UniItem so eviction can work properly

Reply:
One thing to note is that because this isn't dependent upon the bulk indexing operation, the value will be in the cache before the index update occurs. This is probably fine as it reduces a lot of allocations though.